### ISSUE TITLE
Javascript support

### ISSUE THREAD

**Issue by miah** [dbrady/tourbus#14]
"@miah_ For me the biggest stumbling block has been javascript support. I think that would bring TB into the 20-teens decade."

**Comment by xunker** [comment]
I've looked in to adding this a few times myself, and it seems the root difficultly is that TB is based on mechanize which has no provisions for JS. The logical solution is to move TB to something modern like Capybara.

Problem there is that since TB is really just a wrapper for Webrat/Mechanize calls it could break many/most/all existing TB scripts. There would need to be a hard version break, and tours would need to be updated before use on the new branch.

(In no way should this sound like a cop-out or a WONTFIX, but instead as a "this is pretty hard.")

**Comment by dbrady** [comment]
This is definitely the feature that would make me yell at people to start using TourBus. Right now the only way I have seen to get full javascript support is to go with selenium, which makes it problematic to a) run 100 tourbusses on a single laptop and b) run a tourbus from a headless linux box.

There's a selenium headless out there now, which might be worth investigating; I think all it does is run firefox or IE pointed at a virtual Xframebuffer though, which means again it might be hard to spin up 100 of them simultaneously on a single machine.

**Comment by dbrady** [comment]
@xunker Agreed -- I think this level of change would be worthy of a "3.0" version change so people who have it working the way they like would still have that.

**Comment by stevemadere** [comment]
OK so now that phantomjs is an option for browser emulation with javascript execution, are there other alternative projects to tourbus that are doing a similar thing using that?
